Api/V8/Service/ModuleService.php

```diff
@@ -113,39 +116,92 @@ public function getRecords(GetModulesParams $params, Request $request)
         $realRowCount = $this->beanManager->countRecords($module, $where);
         $limit = $size === BeanManager::DEFAULT_ALL_RECORDS ? BeanManager::DEFAULT_LIMIT : $size;
         $deleted = $params->getDeleted();
+        $data = [];
 
         if (empty($fields)) {
             $fields = $this->beanManager->getDefaultFields($bean);
         }
 
-        $beanListResponse = $this->beanManager->getList($module)
-            ->orderBy($orderBy)
-            ->where($where)
-            ->offset($offset)
-            ->limit($limit)
-            ->max($size)
-            ->deleted($deleted)
-            ->fields($this->beanManager->filterAcceptanceFields($bean, $fields))
-            ->fetch();
-
-
-        $beanArray = [];
-        foreach ($beanListResponse->getBeans() as $bean) {
-            $bean = $this->beanManager->getBeanSafe(
-                $params->getModuleName(),
-                $bean->id
-            );
-            $beanArray[] = $bean;
+
+        // Detect if bean has email field
+        if ((property_exists($bean, 'email1') && strpos($where, 'email1')) || (property_exists($bean,
+                    'email2') && strpos($where, 'email2'))) {
+            $selectedFields = strtolower($module) . '.' . implode(',' . substr(strtolower($module), 0, 1) . '.', $fields);
```

Comment:
Couple changes I'd make here:

String normalisation:
```php
strtolower(substr($module, 0, 1))
```
Fixing potential invalid index:

```php
strtolower($module[0])
```

Which should give us:
```suggestion
$selectedFields = strtolower($module) . '.' . implode(',' . strtolower($module[0]) . '.', $fields);
```

Although actually looking at this, can this var just be removed? I don't see it actually being used anywhere.